### PR TITLE
chore: add crd description for Promise

### PR DESCRIPTION
**Pull request by ChunyiLyu**
## Context

Adding crd descriptions for Promise.

Description mostly taken from https://docs.kratix.io/main/guides/writing-a-promise#whats-inside-a-kratix-promise.

Relates to chore #106

### CODE REVIEW

File: config/crd/bases/platform.kratix.io_promises.yaml [hunk 1]
```diff
@@ -55,17 +55,23 @@ spec:
...
+                  API an application developers will use to request a Resource from this Promise.
+                  Must be a valid kubernetes custom resource definition.
                 type: object
                 x-kubernetes-embedded-resource: true
                 x-kubernetes-preserve-unknown-fields: true
               dependencies:
+                description: A collection of prerequisites that enable the creation
+                  of a Resource.
                 items:
                   description: Resources represents the manifest workload to be deployed
```
Comment (SaphMB) [inline config/crd/bases/platform.kratix.io_promises.yaml:68]:
```suggestion
                  description: Resources represent the manifest workload to be deployed
```

File: config/crd/bases/platform.kratix.io_promises.yaml [hunk 2]
```diff
@@ -55,17 +55,23 @@ spec:
...
                     on Destinations
                   type: object
                   x-kubernetes-preserve-unknown-fields: true
                 type: array
               destinationSelectors:
+                description: A list of key and value pairs (labels) used for scheduling.
                 items:
                   description: For Promise spec
```
Comment (SaphMB) [inline config/crd/bases/platform.kratix.io_promises.yaml:76]: Not sure if this is needed

```
GROUP:      platform.kratix.io
KIND:       Promise
VERSION:    v1alpha1

FIELD: destinationSelectors <[]Object>


DESCRIPTION:
    A list of key and value pairs (labels) used for scheduling.
    For Promise spec

FIELDS:
  matchLabels	<map[string]string>
    <no description>
```

File: config/crd/bases/platform.kratix.io_promises.yaml [hunk 3]
```diff
@@ -76,6 +82,9 @@ spec:
                   type: object
                 type: array
               requiredPromises:
+                description: |-
+                  A list of Promises that are required by this Promise.
+                  All required Promises must be present and available for this promise to be made available.
```
Comment (SaphMB) [inline config/crd/bases/platform.kratix.io_promises.yaml:87]:
```suggestion
                  All required Promises must be present and available for this Promise to be made available.
```

File: api/v1alpha1/promise_types.go [hunk 4]
```diff
@@ -40,30 +40,27 @@ const (
...
 
-	// TODO (has since been merged!): apiextension.CustomResourceDefinitionSpec struct(s) don't have the required jsontags and
-	// cannot be used as a Type. See https://github.com/kubernetes-sigs/controller-tools/pull/528
-	// && https://github.com/kubernetes-sigs/controller-tools/issues/291
-	//
-	// OPA Validation pattern:
-	// https://github.com/open-policy-agent/frameworks/blob/1307ba72bce38ee3cf44f94def1bbc41eb4ffa90/constraint/pkg/apis/templates/v1beta1/constrainttemplate_types.go#L46
-	// API runtime.RawExtension      `json:"api,omitempty"`
-
+	// API an application developers will use to request a Resource from this Promise.
```
Comment (SaphMB) [inline api/v1alpha1/promise_types.go:46]:
```suggestion
	// API application developers will use to request a Resource from this Promise.
```

**Review by SaphMB — COMMENTED** [review]
Looks good! Just a couple of small suggestions

File: config/crd/bases/platform.kratix.io_promises.yaml (hunk 2 above)
Comment (ChunyiLyu) [reply config/crd/bases/platform.kratix.io_promises.yaml]: Good spot! Removed :)

File: config/rbac/role.yaml [hunk 5]
```diff
@@ -28,7 +28,7 @@ rules:
   verbs:
   - get
   - list
-  - watch
+  - watch // Also used by promiseRelease
```
Comment (aclevername) [inline config/rbac/role.yaml:31]: rebase main, this should go away

**Review by aclevername — APPROVED** [review]
LGTM, 1 blocker see comment

**Pull request merged**